### PR TITLE
Issue/dialog glass pane

### PR DESCRIPTION
**Pull request by danielwiehl**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [guidelines](https://github.com/SchweizerischeBundesbahnen/scion-workbench/blob/master/CONTRIBUTING.md)
- [x] Tests for the changes have been added
- [x] Docs have been added or updated

## PR Type
What kind of change does this PR introduce?

- [x] Fix
- [ ] Feature
- [ ] Documentation
- [ ] Refactoring (changes that neither fixes a bug nor adds a feature)
- [ ] Performance (changes that improve performance)
- [ ] Test (adding missing tests, refactoring tests; no production code change)
- [ ] Chore (other changes like formatting, updating the license, removal of deprecations, etc)
- [ ] Deps (changes related to updating dependencies)
- [ ] CI (changes to our CI configuration files and scripts)
- [ ] Revert (revert of a previous commit)
- [ ] Release (publish a new release)
- [ ] Other... Please describe:

## Issue

Issue Number: #

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/scion/e2e-testing/src/dialog.po.ts [hunk 1]
```diff
@@ -46,8 +46,15 @@ export class DialogPO {
     return fromRect(await this._dialog.boundingBox());
   }
 
-  public async getGlassPaneBoundingBox(): Promise<DomRect> {
-    return fromRect(await this.locator.page().locator('.cdk-overlay-pane.wb-dialog-glass-pane', {has: this.locator}).boundingBox());
+  public async getGlassPaneBoundingBoxes(): Promise<Set<DomRect>> {
+    const dialogId = await this.locator.getAttribute('data-dialogid');
+    const glassPaneLocators = await this.locator.page().locator(`div.e2e-glasspane[data-owner="${dialogId}"]`).all();
+
+    const boundingBoxes = new Set<DomRect>();
```
Comment (k-genov) [inline projects/scion/e2e-testing/src/dialog.po.ts:53]: Why Set?

File: projects/scion/e2e-testing/src/workbench/popup.e2e-spec.ts [hunk 2]
```diff
@@ -541,6 +541,55 @@ test.describe('Workbench Popup', () => {
       await expect(await popup.isPresent()).toBe(true);
       await expect(await popup.isVisible()).toBe(true);
     });
+
+    test('should propagate view context', async ({appPO, workbenchNavigator}) => {
+      await appPO.navigateTo({microfrontendSupport: false});
+
+      // Open target view.
+      const popupTagetViewPage = await workbenchNavigator.openInNewTab(ViewPagePO);
```
Comment (k-genov) [inline projects/scion/e2e-testing/src/workbench/popup.e2e-spec.ts:549]:
```suggestion
      const popupTargetViewPage = await workbenchNavigator.openInNewTab(ViewPagePO);
```

File: projects/scion/e2e-testing/src/workbench/popup.e2e-spec.ts [hunk 3]
```diff
@@ -541,6 +541,55 @@ test.describe('Workbench Popup', () => {
...
+      // Open popup in target view.
+      const popupOpenerPage1 = await workbenchNavigator.openInNewTab(PopupOpenerPagePO);
+      await popupOpenerPage1.selectPopupComponent('popup-opener-page');
+      await popupOpenerPage1.enterCssClass('testee-1');
+      await popupOpenerPage1.enterCloseStrategy({closeOnFocusLost: false});
+      await popupOpenerPage1.enterContextualViewId(popupTagetViewPage.viewId);
+      await popupOpenerPage1.enterPosition({left: 300, top: 300});
+      await popupOpenerPage1.clickOpen();
+
+      const popup1 = appPO.popup({cssClass: 'testee-1'});
```
Comment (k-genov) [inline projects/scion/e2e-testing/src/workbench/popup.e2e-spec.ts:560]: The initial assertions for popup1 should already be covered by a dedicated test for view context?

File: projects/scion/workbench/src/lib/common/dom.util.ts [hunk 4]
```diff
@@ -32,6 +33,22 @@ export function setStyle(element: HTMLElement | ElementRef<HTMLElement>, styles:
   });
 }
 
+/**
+ * Sets the given attribute(s) on the given element.
+ * To remove an attribute, set its value to `null`.
+ */
+export function setAttribute(element: HTMLElement | ElementRef<HTMLElement>, attributes: {[name: string]: any | null}): void {
```
Comment (k-genov) [inline projects/scion/workbench/src/lib/common/dom.util.ts:40]:
```suggestion
export function setAttribute(element: HTMLElement | ElementRef<HTMLElement>, attributes: {[name: string]: unknown | null}): void {
```
?

File: projects/scion/workbench/src/lib/dialog/ɵworkbench-dialog.ts [hunk 5]
```diff
@@ -74,6 +86,7 @@ export class ɵWorkbenchDialog<R = unknown> implements WorkbenchDialog<R> {
     this.blockWhenNotOnTop();
     this.restoreFocusOnAttach();
     this.restoreFocusOnUnblock();
+    this.blinkWhenRequested();
```
Comment (k-genov) [inline projects/scion/workbench/src/lib/dialog/ɵworkbench-dialog.ts:89]:
```suggestion
    this.blinkOnRequested();
```

File: projects/scion/workbench/src/lib/dialog/ɵworkbench-dialog.ts [hunk 6]
```diff
@@ -314,10 +334,23 @@ export class ɵWorkbenchDialog<R = unknown> implements WorkbenchDialog<R> {
...
       .pipe(
-        map(top => top !== this),
+        map(top => top === this ? null : top),
+        takeUntilDestroyed(this._destroyRef),
+      )
+      .subscribe(this.blockedBy$);
+  }
+
+  /**
+   * Blinks briefly when requested.
```
Comment (k-genov) [inline projects/scion/workbench/src/lib/dialog/ɵworkbench-dialog.ts:344]: Do we need the comment? It doesn't give more information than the name, does it?

File: projects/scion/e2e-testing/src/dialog.po.ts (hunk 1 above)
Comment (danielwiehl) [reply projects/scion/e2e-testing/src/dialog.po.ts:53]: We do not know the order of the glass panes.

File: projects/scion/e2e-testing/src/workbench/popup.e2e-spec.ts (hunk 3 above)
Comment (danielwiehl) [reply projects/scion/e2e-testing/src/workbench/popup.e2e-spec.ts:560]: I know, but should also be tested in the context of this test.

File: projects/scion/workbench/src/lib/common/dom.util.ts (hunk 4 above)
Comment (danielwiehl) [reply projects/scion/workbench/src/lib/common/dom.util.ts:40]: Should be a `string`. I also changed the other occurrences (`setStyle`, `setCssVariable`).

File: projects/scion/workbench/src/lib/popup/popup.service.ts [hunk 7]
```diff
@@ -239,13 +245,13 @@ export class PopupService {
    * Hides the popup when its contextual view is detached, and displays it when it is reattached. Also restores the focus on re-activation.
    * The contextual view is detached if not active, or located in the peripheral area and the main area is maximized.
    */
-  private hidePopupOnViewDetach(overlayRef: OverlayRef, contextualView: ɵWorkbenchView, takeUntilClose: <T>() => MonoTypeOperatorFunction<T>): void {
+  private hidePopupOnViewDetach(overlayRef: OverlayRef, contextualView: ɵWorkbenchView, popup: ɵPopup): void {
```
Comment (k-genov) [inline projects/scion/workbench/src/lib/popup/popup.service.ts:248]: Pass only the destroyRef as argument?

File: projects/scion/workbench/src/lib/workbench.component.html [hunk 8]
```diff
@@ -1,21 +1,18 @@
...
+    @if (whenViewContainersInjected | async) {
+      <wb-workbench-layout/>
+    }
+    <!-- Achor to attach iframes of embedded microfrontends -->
+    <ng-container #iframe_host/>
   </div>
 
-  <!-- anchor to attach the visual placeholder when dragging a view over a valid drop zone -->
+  <!-- Anchor to attach the visual placeholder when dragging a view over a valid drop zone -->
   <div #view_drop_placeholder_host class="view-drop-placeholder-host"></div>
```
Comment (k-genov) [inline projects/scion/workbench/src/lib/workbench.component.html:11]:
```suggestion
  <div #view_drop_placeholder_host class="view-drop-placeholder-host"/>
```

**Review by k-genov — APPROVED** [review]
lgtm

**Pull request merged**